From 84edfafefe10dd2d9a3ab93202050a551f023016 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 1 Jun 2016 14:34:42 +0100
Subject: [PATCH] shobjidl: Fix compile-guard around SHARDAPPIDINFOLINK

Both MSVC and mingw-w64 define NTDDI_VERSION and NTDDI_WIN7 and both
expose the SHARDAPPIDINFOLINK for >= NTDDI_WIN7, so only define it
when targeting eariler Windows SDKs.

Change-Id: I5bf297bfc5624d98838e48288c5139acf13fc431
---
 src/winextras/winshobjidl_p.h | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git qtwinextras/src/winextras/winshobjidl_p.h qtwinextras/src/winextras/winshobjidl_p.h
index 224150d..c0893f8 100644
--- qtwinextras/src/winextras/winshobjidl_p.h
+++ qtwinextras/src/winextras/winshobjidl_p.h
@@ -223,9 +223,7 @@ public:
 
 #endif
 
-#if (defined _MSC_VER && _MSC_VER < 1600) || defined(Q_CC_MINGW)
-
-#   if !defined(__MINGW64_VERSION_MAJOR) || !defined(__MINGW64_VERSION_MINOR) || __MINGW64_VERSION_MAJOR * 100 + __MINGW64_VERSION_MINOR < 301
+#if (NTDDI_VERSION < NTDDI_WIN7)
 
 typedef struct SHARDAPPIDINFOLINK
 {
@@ -235,8 +233,6 @@ typedef struct SHARDAPPIDINFOLINK
     PCWSTR pszAppID;        // The id of the application that should be associated with this recent doc.
 } SHARDAPPIDINFOLINK;
 
-#   endif // !defined(__MINGW64_VERSION_MAJOR) || !defined(__MINGW64_VERSION_MINOR) || __MINGW64_VERSION_MAJOR * 100 + __MINGW64_VERSION_MINOR < 301
-
 #endif
 
 #endif // WINSHOBJIDL_P_H
-- 
2.10.2

